mlflow/store/dbfs_artifact_repo.py

```diff
@@ -28,10 +30,38 @@ class DbfsArtifactRepository(ArtifactRepository):
     def __init__(self, artifact_uri):
         cleaned_artifact_uri = artifact_uri.rstrip('/')
         super(DbfsArtifactRepository, self).__init__(cleaned_artifact_uri)
+        if mlflow.utils.databricks_utils.is_dbfs_fuse_available() \
```

Comment:
Can we move this logic to a factory method returning the appropriate instance of artifact repository instead? I think it would be cleaner and easier to maintain than having nested repo and reroute methods to it.

Reply:
Yep we could (e.g. in https://github.com/mlflow/mlflow/blob/master/mlflow/store/artifact_repository_registry.py#L51) - I just wanted to avoid adding code specific to the DBFS case to the factory method

Reply:
I was thinking if we could register dbfs_artifact_repo_factory as a handler for "dbfs"